src/vpic/dump.cc

```diff
@@ -853,3 +889,773 @@ vpic_simulation::hydro_dump( const char * speciesname,
+    // Convert Cell ID to global
+    int64_t local_i = sp->p[n].i;
+
+    int ix, iy, iz, rx, ry, rz;
+    // Convert rank to local x/y/z
+    UNVOXEL(mpi_rank, rx, ry, rz, grid->gpx, grid->gpy, grid->gpz);
+
+    // Calculate local ix/iy/iz
+    UNVOXEL(local_i, ix, iy, iz, grid->nx+2, grid->ny+2, grid->nz+2);
+
+    // Account for the "first" ghost cell
+    ix = ix - 1;
+    iy = iy - 1;
+    iz = iz - 1;
+
+    // Convert ix/iy/iz to global
+    int gix = ix + (grid->nx * rx);
+    int giy = iy + (grid->ny * ry);
+    int giz = iz + (grid->nz * rz);
+
+    // calculate global grid sizes
+    int gnx = grid->nx * grid->gpx;
+    int gny = grid->ny * grid->gpy;
+    int gnz = grid->nz * grid->gpz;
+
+    // TODO: find a better way to account for the hard coded ghosts in VOXEL
+    int64_t global_i = VOXEL(gix, giy, giz, gnx-2, gny-2, gnz-2);
+    #endif
+
+    const size_t index = frame*sp->buf_n_particles + n;
+
+    // Particle Properties
+    sp->output_buffer_dx[index] = sp->p[n].dx;
+    sp->output_buffer_dy[index] = sp->p[n].dy;
+    sp->output_buffer_dz[index] = sp->p[n].dz;
+
+    #ifdef OUTPUT_CONVERT_GLOBAL_ID
+    sp->output_buffer_i[index]  = global_i;
+    #else
+    sp->output_buffer_i[index]  = sp->p[n].i;
+    #endif
+
+    sp->output_buffer_ux[index] = sp->p[n].ux;
+    sp->output_buffer_uy[index] = sp->p[n].uy;
+    sp->output_buffer_uz[index] = sp->p[n].uz;
+    sp->output_buffer_w[index]  = sp->p[n].w;
+
+    #ifdef VPIC_GLOBAL_PARTICLE_ID
+    if(sp->has_ids) {
+      sp->output_buffer_id[index] = sp->p_id[n];
+    }
+    #endif
+
+    // Annotations
+    #ifdef VPIC_PARTICLE_ANNOTATION
+    if(sp->has_annotation) {
+      for(int a = 0; a < sp->has_annotation; a++) {
+       const size_t out_index = sp->buf_n_particles * sp->buf_n_frames * a + sp->buf_n_particles * frame + n;
+       const size_t in_index  = sp->has_annotation * n + a;
+       sp->output_buffer_an[out_index] = sp->p_annotation[in_index];
+      }
+    }
+    #endif
+
+    sp->output_buffer_ts[index] = step();
+    sp->buf_n_valid[frame]++;
+  }
+}
+
+#undef UNVOXEL
+
+/**
+ * @brief Writes the accumulated buffer to disk
+ *
+ * @param sp_name Name of the species. Should be uniq and should exist
+ *
+ * @return Nothing, but performs IO
+ */
+void
+vpic_simulation::write_buffered_particle_dump(const char * sp_name) {
+  species_t *sp = find_species_name( sp_name, species_list );
+  if( !sp ) ERROR(( "Invalid species \"%s\"", sp_name ));
+
+#ifdef VPIC_ENABLE_HDF5
+  char strbuf[4096];
+
+  // MPI-Info object. necessary to create the property list for parallel file
+  // open and to set IO hints
+  MPI_Info info;
+  MPI_Info_create(&info);
+  MPI_Info_set(info, "romio_cb_read", "automatic");
+  MPI_Info_set(info, "romio_cb_write", "automatic");
+  hid_t pl_file = H5Pcreate(H5P_FILE_ACCESS);
+  H5Pset_fapl_mpio(pl_file, MPI_COMM_WORLD, info);
+  hid_t pl_link = H5Pcreate(H5P_LINK_ACCESS);
+  MPI_Info_free(&info);
+
+  // Open file
+  dump_mkdir("tracer");
```

Comment:
Do we expect a user might want to be able to configure this?

Reply:
The user is now afforded some extra configurability (the directory name). Not sure what a sane interface that gives much more freedom would look like.